app/tests/test_search_utils.py

```diff
@@ -0,0 +1,311 @@
+            [{"_source": {"event_date": "30/10/2024"}}],
+        ),
+        # multiple dates in the correct format
+        (
+            [
+                {
+                    "_source": {
+                        "event_date": "2024-10-30T13:45:30",
+                        "creation_date": "2023-08-20T07:15:00",
+                    }
+                }
+            ],
+            [
+                {
+                    "_source": {
+                        "event_date": "30/10/2024",
+                        "creation_date": "20/08/2023",
+                    }
+                }
+            ],
+        ),
+        # no date field
+        (
+            [
+                {
+                    "_source": {
+                        "name": "Test Event",
+                        "description": "An event description",
+                    }
+                }
+            ],
+            [
+                {
+                    "_source": {
+                        "name": "Test Event",
+                        "description": "An event description",
+                    }
+                }
+            ],
+        ),
+        # empty date string
+        ([{"_source": {"event_date": ""}}], [{"_source": {"event_date": ""}}]),
+        # invalid date format (should return the original string)
+        (
+            [{"_source": {"event_date": "30-10-2024 13:45"}}],
+            [{"_source": {"event_date": "30-10-2024 13:45"}}],
+        ),
+        # empty results list
+        ([], []),
+    ],
+)
+def test_format_opensearch_results(results, expected):
+    actual = format_opensearch_results(results)
+    assert actual == expected
+
+
+@pytest.mark.parametrize(
+    "search_area, expected_fields",
+    [
+        # default "all" fields (no specific search area)
+        ("all", ["*"]),
+        # "metadata" search area
+        (
+            "metadata",
+            ["metadata_field_1", "metadata_field_2", "metadata_field_3"],
+        ),
+        # "record" search area
+        ("record", ["file_name", "file_path", "content"]),
+        # empty search_area string (should default to "all" behavior)
+        ("", ["*"]),
+        # none as search_area (should default to "all" behavior)
+        (None, ["*"]),
+        # invalid search_area (should default to "all" behavior)
+        ("invalid_area", ["*"]),
+    ],
+)
+@patch("app.main.util.search_utils.OpenSearch")
+@patch("app.main.util.search_utils.get_all_fields_excluding")
+def test_get_open_search_fields_to_search_on(
+    mock_get_all_fields_excluding, mock_opensearch, search_area, expected_fields
+):
+    mock_get_all_fields_excluding.return_value = [
+        "metadata_field_1",
+        "metadata_field_2",
+        "metadata_field_3",
+    ]
+    actual_fields = get_open_search_fields_to_search_on(
+        mock_opensearch, search_area
+    )
+    assert actual_fields == expected_fields
+
+
+def test_get_param(app):
+    with patch(
```

Comment:
If we make this functional like the rest of the functions, we can pass request in and then tetsing we can pass in an object with the same interface as request